### DISCUSSION TITLE
Run the bot on pythonanywhere.com

### DISCUSSION THREAD

**Discussion by ghost** [n3d1117/chatgpt-telegram-bot#143] [Ideas]
I'm trying to run the bot on PythonAnywhere , and it kind of works , the bot is online and responds if i write to it from telegrtam   but there is an error about connecting to the openAI API , it's a strange error  . 
My question is , how could i make the bot work online without me hosting the server on my computer .

**Comment by n3d1117** [comment]
Hi @adrianplanet, I think free accounts on PythonAnywhere have outbound internet access limited by a [whitelist](https://www.pythonanywhere.com/whitelist/), see their [pricing page](https://www.pythonanywhere.com/pricing/).

Maybe you can try [heroku](https://www.heroku.com/) or [fly.io](https://fly.io/)
